internal/controller/datadogagent/feature/appsec/feature_test.go

```diff
@@ -0,0 +1,623 @@
+			annotations: map[string]string{
+				AnnotationInjectorEnabled:    "true",
+				AnnotationInjectorAutoDetect: "true",
+			},
+			wantConfig: Config{
+				Enabled:    true,
+				AutoDetect: boolPtr(true),
+			},
+			wantErr: false,
+		},
+		{
+			name: "enabled with proxies",
+			annotations: map[string]string{
+				AnnotationInjectorEnabled: "true",
+				AnnotationInjectorProxies: `["envoy-gateway","istio"]`,
+			},
+			wantConfig: Config{
+				Enabled: true,
+				Proxies: []string{"envoy-gateway", "istio"},
+			},
+			wantErr: false,
+		},
+		{
+			name: "enabled with processor port",
+			annotations: map[string]string{
+				AnnotationInjectorEnabled:       "true",
+				AnnotationInjectorProcessorPort: "443",
+			},
+			wantConfig: Config{
+				Enabled:       true,
+				ProcessorPort: 443,
+			},
+			wantErr: false,
+		},
+		{
+			name: "invalid enabled value",
+			annotations: map[string]string{
+				AnnotationInjectorEnabled: "invalid",
+			},
+			wantErr: true,
+		},
+		{
+			name: "invalid autoDetect value",
+			annotations: map[string]string{
+				AnnotationInjectorEnabled:    "true",
+				AnnotationInjectorAutoDetect: "invalid",
+			},
+			wantErr: true,
+		},
+		{
+			name: "invalid proxies JSON",
+			annotations: map[string]string{
+				AnnotationInjectorEnabled: "true",
+				AnnotationInjectorProxies: "not-json",
+			},
+			wantErr: true,
+		},
+		{
+			name: "invalid processor port",
+			annotations: map[string]string{
+				AnnotationInjectorEnabled:       "true",
+				AnnotationInjectorProcessorPort: "not-a-number",
+			},
+			wantErr: true,
+		},
+		{
+			name: "full config",
+			annotations: map[string]string{
+				AnnotationInjectorEnabled:                   "true",
+				AnnotationInjectorAutoDetect:                "false",
+				AnnotationInjectorProxies:                   `["envoy-gateway"]`,
+				AnnotationInjectorProcessorPort:             "8080",
+				AnnotationInjectorProcessorAddress:          "processor.example.com",
+				AnnotationInjectorProcessorServiceName:      "appsec-svc",
+				AnnotationInjectorProcessorServiceNamespace: "datadog",
+			},
+			wantConfig: Config{
+				Enabled:                   true,
+				AutoDetect:                boolPtr(false),
+				Proxies:                   []string{"envoy-gateway"},
+				ProcessorPort:             8080,
+				ProcessorAddress:          "processor.example.com",
+				ProcessorServiceName:      "processor.example.com",
+				ProcessorServiceNamespace: "processor.example.com",
+			},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			config, err := FromAnnotations(tt.annotations)
+
+			if tt.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.wantConfig.Enabled, config.Enabled)
+				assert.Equal(t, tt.wantConfig.AutoDetect, config.AutoDetect)
+				assert.Equal(t, tt.wantConfig.Proxies, config.Proxies)
+				assert.Equal(t, tt.wantConfig.ProcessorPort, config.ProcessorPort)
```

Comment:
I think there are missing assertions for `ProcessorAddress`, `ProcessorServiceName`, `ProcessorServiceNamespace`, since in the desired config they don't correspond to the annotations added (ex: full config test)